src/pyodide.js

```diff
@@ -361,6 +361,95 @@ var languagePluginLoader = new Promise((resolve, reject) => {
     return Module.runPython(code);
   };
 
+  Module.function_supports_kwargs = function(funcstr) {
```

Comment:
Do we have to re-implement this ourselves? Is no code we could just vendor here, just so that we don't have to extensively check all corner cases?

Reply:
Good question. I will look on npm. I crowdsourced this a bit by posting it on codegolf:
https://codegolf.stackexchange.com/questions/217511/does-this-javascript-function-support-keyword-arguments
See in particular the one javascript answer:
https://codegolf.stackexchange.com/a/217518/77634
They use `Reflect.parse` which is a `SpiderMonkey` API absent from web standard.

Once we tranform `pyodide.js` into a npm package I think we should use this:
https://www.npmjs.com/package/ast-types

Reply:
The codegolf folks found exactly one edge case that defeated my implementation: that is if someone defines a function like this:
`function f(...[x, y, { option1, option2}]){}`. This function technically "supports keyword arguments" but my function gives a false negative. However, nobody defined functions like that.

Reply:
I think given that I have a large number of tests for this and both my test suite and my implementation (minus string support) was vetted by the five people who answered the question on codegolf, it would be reasonable to add this for now.

What do you think?

Reply:
@dalcde Do you have an opinion?

Reply:
I'm not opposed to include it if there are no alternatives.

What about performance? I see you cache it, but I imagine there are use cases where one would continuously create new PyProxy objects in a loop and call them? What would be the performance impact of this check?

Reply:
> What would be the performance impact of this check?

I don't think the performance impact is very high:

1. The check never happens unless you try to call a javascript function with a keyword argument
2. The performance cost: first it copies the string representation of the function, then it scans the string representation doing a small number of comparisons per character. If you are willing to copy the string a second time, I could implement the same algorithm in C, but this sort of code likely JITs extremely well so it's probably better to leave it alone.